File: languages/django_models/species.py
from .abstract_element_model import AbstractElementModel
from django.core.validators import MaxValueValidator
from django.db import models

class Species(AbstractElementModel):

    # Biology
    appearance = models.TextField(blank=True, null=True)
    life_span = models.PositiveIntegerField(blank=True, null=True)
    typical_weight = models.PositiveIntegerField(blank=True, null=True)
    diet = models.ManyToManyField("Species", blank=True, related_name="species_diet")
    reproduction = models.ManyToManyField("Construct", blank=True, related_name="species_reproduction")

    # Psychology
    instincts = models.TextField(blank=True, null=True)
    sociality = models.TextField(blank=True, null=True)
    temperament = models.TextField(blank=True, null=True)
    communication = models.TextField(blank=True, null=True)
    aggression = models.PositiveIntegerField(validators=[MaxValueValidator(100)], blank=True, null=True)
    traits = models.ManyToManyField("Trait", blank=True, related_name="species_traits")

    # World
    role = models.TextField(blank=True, null=True)
    parent_species = models.ForeignKey("Species", on_delete=models.SET_NULL, blank=True, null=True, related_name="species_parent_species")
    locations = models.ManyToManyField("Location", blank=True, related_name="species_locations")
    zones = models.ManyToManyField("Zone", blank=True, related_name="species_zones")
    affinities = models.ManyToManyField("Phenomenon", blank=True, related_name="species_affinities")

    def __str__(self):
        return self.name

File: languages/csharp_unity_json_properties/Species.cs
using System;
using System.Collections.Generic;
using Newtonsoft.Json;
using Utils;

[System.Serializable]
public class Species : Element
{
    [JsonProperty("appearance"), TextAttribute("")]
    public string appearance;
    [JsonProperty("life_span"), Integer(0)]
    public int lifeSpan;
    [JsonProperty("weight"), Integer(0)]
    public int weight;
    [JsonProperty("nourishment"), ReferenceAttribute(typeof(Species), true)]
    public string nourishment;
    [JsonProperty("reproduction"), ReferenceAttribute(typeof(Construct), true)]
    public string reproduction;
    [JsonProperty("adaptations"), ReferenceAttribute(typeof(Ability), true)]
    public string adaptations;
    [JsonProperty("instincts"), TextAttribute("")]
    public string instincts;
    [JsonProperty("sociality"), TextAttribute("")]
    public string sociality;
    [JsonProperty("temperament"), TextAttribute("")]
    public string temperament;
    [JsonProperty("communication"), TextAttribute("")]
    public string communication;
    [JsonProperty("aggression"), Integer(100)]
    public int aggression;
    [JsonProperty("traits"), ReferenceAttribute(typeof(Trait), true)]
    public string traits;
    [JsonProperty("role"), TextAttribute("")]
    public string role;
    [JsonProperty("parent_species"), ReferenceAttribute(typeof(Species))]
    public string parentSpecies;
    [JsonProperty("locations"), ReferenceAttribute(typeof(Location), true)]
    public string locations;
    [JsonProperty("zones"), ReferenceAttribute(typeof(Zone), true)]
    public string zones;
    [JsonProperty("affinities"), ReferenceAttribute(typeof(Phenomenon), true)]
    public string affinities;
}
